injectproxy/routes.go

```diff
@@ -118,27 +120,65 @@ func (r *routes) noop(w http.ResponseWriter, req *http.Request) {
 }
 
 func (r *routes) query(w http.ResponseWriter, req *http.Request) {
-	expr, err := parser.ParseExpr(req.FormValue("query"))
+	// The `query` can come in the URL query string and/or the POST body.
+	// For this reason, we need to try to enforcing in both places.
+	// Note: a POST request may include some values in the URL query string
+	// and others in the body. If both locations include a `query`, then
+	// enforce in both places.
+	q, f1, err := enforceValues(req.Context(), r.label, req.URL.Query())
 	if err != nil {
 		return
```

Comment:
all of those errors I think should be reporter somehow :thinking:

Reply:
We don't do this before. I would prefer to introduce this in a followup

Reply:
sgtm